### PR TITLE
Fix: Handling multi-path KUBECONFIG and updating kubectl download URLs

### PR DESCRIPTION
**Pull request by tejhan**
This PR fixes KUBECONFIG handling when set to colon-separated paths (e.g. `/path1:/path2`). This should several of the partial breaks we have throughout the extension for users with multiple kubeconfig paths. Additionally, our kubectl download links have been updated, which should also fix errors observed when installing kubectl via the extension for later kubectl versions (>v1.31)

Functions updated:

- **loadKubeconfig()**: Split multi-path and merge configs sequentially
- **mergeToKubeconfig()**: Write to first path only
- **getCredentials()**: Pass first path to az CLI
- **getKubeconfigPathHash()**: Updated path existence check to loop for multi-path cases 

Additional Changes: 
- **kubectl download URLs**: Updated from deprecated `storage.googleapis.com` to `dl.k8s.io` (fixes 404s for kubectl > v1.31)
- **downloadKubectlVersion()**: Only chmod after successful download (fixes ENOENT errors)

Related issues: 
- https://github.com/vscode-kubernetes-tools/vscode-kubernetes-tools/issues/1733

### CODE REVIEW

File: src/components/kubectl/autoversion.ts [hunk 1]
```diff
@@ -117,7 +120,16 @@ async function getKubeconfigPathHash(): Promise<string | undefined> {
 
     // TODO: fs.existsAsync looks in the host filesystem, even in the case of a WSL path. This needs fixing to handle
     // WSL paths properly.
-    if (!await fs.existsAsync(kubeconfigFilePath)) {
+    // For multi-path KUBECONFIG, split and check for path existence
+    const paths = kubeconfigFilePath.split(path.delimiter).filter((p) => p.length > 0);
```
[the remark this answers is not in the document]
Comment (tejhan) [reply src/components/kubectl/autoversion.ts]: This is referring to kubeconfigPath.pathType === "wsl", which is only invoked for users with the `use-wsl` setting, which is outdated & currently broken/ not widely applicable due to VScode's remote WSL support now. Will be refactoring for these in the future.

File: src/components/kubectl/kubeconfig.ts [hunk 2]
```diff
@@ -36,7 +36,25 @@ export async function loadKubeconfig(): Promise<any> {
...
+        const currentContext = kubeconfig.getCurrentContext();
+        // Merge any additional files
+        for (let i = 1; i < paths.length; i++) {
+            const kc = new kubernetes.KubeConfig();
+            kc.loadFromFile(paths[i]);
+            kubeconfig.mergeConfig(kc);
+        }
+        // Restore the original current context
+        if (currentContext) {
+            kubeconfig.setCurrentContext(currentContext);
```
[the remark this answers is not in the document]
Comment (tejhan) [reply src/components/kubectl/kubeconfig.ts]: Copilot is being overly defensive here & is suggesting a scenario that won't occur. Contexts won't be overwritten with `mergeConfig` - even if duplicates exist, they're appended, not overwritten.

File: src/components/kubectl/autoversion.ts [hunk 3]
```diff
@@ -78,16 +78,19 @@ async function downloadKubectlVersion(shell: Shell, host: Host, serverVersion: s
...
         download.once(kubectlUrl, binPath)
     );
+    if (!succeeded(downloadResult)) {
+        return false;
+    }
     if (shell.isUnix()) {
         await fs.chmod(binPath, '0755');
     }
-    return succeeded(downloadResult);
+    return true;
```
[the remark this answers is not in the document]
Comment (tejhan) [reply src/components/kubectl/autoversion.ts]: Yes, the reason I added the change on 87-89 is that without it, we would execute `fs.chmod` on line 91 even if `downloadResult` failed on line 84.

This means that if the dowload failed, we would still subsequently run chmod on a non-existent location, which could propagate further errors. 

This gate added immedeatly terminates the function if the download is unsuccessful & makes it so that if the code does reach 93, we already know it succeeded, thus we just return `true`.

So really this is to account for lines 90-92 which are necessary logic for executable permissions on unix.

[the remark this answers is not in the document]
Comment (Tatsinnit) [reply src/components/kubectl/autoversion.ts]: ❤️ Nice, thank you for this. This is cool, I wonder if this seems like a good opportunity to add some test for this change moving forward, @ashu8912 what are your thoughts and will you be keen to join this discussion? Thanks heaps to both of you @tejhan and @ashu8912 💡 Given this is more controlled change, small test for this contract will be perfect to lock this behaviour and any future changes.

[the remark this answers is not in the document]
Comment (tejhan) [reply src/components/kubectl/autoversion.ts]: Yes @Tatsinnit , currently looking at building up a more comprehensive test suite to cover Kubeconfig scenarios across the board. This one has been locally tested for common edge cases & OS/environment specifics, but we'll be pushing out the test suites in future PR's to keep things standardized. 

Currently working on setting up an architecture & drawing out a plan for how we can broaden these suites in the upcoming future as well.
